packages/video_player/video_player/lib/src/web_vtt.dart

```diff
@@ -0,0 +1,200 @@
+    );
+
+    final String text = captionLines.sublist(hasHeader ? 2 : 1).join('\n');
+
+    // TODO(cyanglaz): Handle special syntax in vtt captions.
+    // https://github.com/flutter/flutter/issues/90007.
+    final String textWithoutFormat = _parseHtmlString(text);
+
+    final Caption newCaption = Caption(
+      number: captionNumber,
+      start: startAndEnd.start,
+      end: startAndEnd.end,
+      text: textWithoutFormat,
+    );
+
+    if (newCaption.start != null && newCaption.end != null) {
+      captions.add(newCaption);
+      captionNumber++;
+    }
+  }
+
+  return captions;
+}
+
+class _StartAndEnd {
+  // When there's an error parsing the start or end, either could be null.
+  final Duration? start;
+  final Duration? end;
+
+  _StartAndEnd(this.start, this.end);
+
+  // Assumes format from an Vtt file.
+  // For example:
+  // 00:09.000 --> 00:11.000
+  static _StartAndEnd fromWebVttString(String line) {
+    final RegExp format =
+        RegExp(_webVttTimeStamp + _webVttArrow + _webVttTimeStamp);
+
+    if (!format.hasMatch(line)) {
+      return _StartAndEnd(null, null);
+    }
+
+    final List<String> times = line.split(_webVttArrow);
+
+    final Duration? start = _parseWebVttTimestamp(times[0]);
+    final Duration? end = _parseWebVttTimestamp(times[1]);
+
+    return _StartAndEnd(start, end);
+  }
+}
+
+String _parseHtmlString(String htmlString) {
+  final Document document = html_parser.parse(htmlString);
+  final Element? body = document.body;
+  if (body == null) {
+    return '';
+  }
+  final Element? bodyElement = html_parser.parse(body.text).documentElement;
+  if (bodyElement == null) {
+    return '';
+  }
+  return bodyElement.text;
+}
+
+// Parses a time stamp in an Vtt file into a Duration.
+// For example:
+//
+// _parseWebVttimestamp('00:01:08.430')
+// returns
+// Duration(hours: 0, minutes: 1, seconds: 8, milliseconds: 430)
+Duration? _parseWebVttTimestamp(String timestampString) {
+  if (!RegExp(_webVttTimeStamp).hasMatch(timestampString)) {
+    return null;
+  }
+
+  final List<String> dotSections = timestampString.split('.');
+  final List<String> hoursMinutesSeconds = dotSections[0].split(':');
+
+  int hours = 0;
+  int minutes = 0;
+  int seconds = 0;
+
+  if (hoursMinutesSeconds.length > 2) {
+    // Timestamp takes the form of [hours]:[minutes]:[seconds].[milliseconds]
+    hours = int.parse(hoursMinutesSeconds[0]);
+    minutes = int.parse(hoursMinutesSeconds[1]);
+    seconds = int.parse(hoursMinutesSeconds[2]);
+  } else if (int.parse(hoursMinutesSeconds[0]) > 59) {
+    // Timestamp takes the form of [hours]:[minutes].[milliseconds]
+    // First position is hours as it's over 59.
+    hours = int.parse(hoursMinutesSeconds[0]);
+    minutes = int.parse(hoursMinutesSeconds[1]);
+  } else {
+    // Timestamp takes the form of [minutes]:[seconds].[milliseconds]
+    minutes = int.parse(hoursMinutesSeconds[0]);
+    seconds = int.parse(hoursMinutesSeconds[1]);
+  }
```

Comment:
Once the above is removed, this can all condense to:
```
int hours = 0;
if (hoursMinutesSeconds.length == 3) {
  hours = int.parse(timeComponents.removeAt(0));
}
final int minutes = int.parse(timeComponents.removeAt(0));
final int seconds = int.parse(timeComponents.removeAt(0));
```

It would be good to add a safety check that the length is either 2 or 3 and returns null first, as well.